plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -237,7 +238,8 @@ protected void refreshFromMetadataLocation(String newLocation)
                             .withMaxRetries(20)
                             .withBackoff(100, 5000, MILLIS, 4.0)
                             .withMaxDuration(Duration.ofMinutes(10))
-                            .abortOn(failure -> failure instanceof ValidationException || isNotFoundException(failure))
+                            .abortOn(ValidationException.class, UnrecoverableS3OperationException.class)
```

Comment:
This exception should be considered private to `TrinoS3FileSystem` (we should make it private) and won't be accessible after we complete the Hadoop removal project. Also, this is only for the legacy S3 file system.

Reply:
should we have some more top level exception for filesystem errors that should not be retried? seems like common requirement, or maybe there is already exception that `UnrecoverableS3OperationException` could extend?

Reply:
We could create a new exception for this, but we should first see if this is applicable to any of the new file systems. Can you investigate that?

Reply:
I have introduced `io.trino.filesystem.UnrecoverableTrinoFileSystemException`, and make this private one extending from it

I can investigate at least some of them, I'll make a list of all implementors of `TrinoFileSystem` that I went / not went through, but let me create another issue/PR for it once this is merged and `io.trino.filesystem.UnrecoverableTrinoFileSystemException` is available

Reply:
I want to make sure that this will be useful in the future and won't become dead code after we remove the legacy S3 file system.